src/js_realm.hpp

```diff
@@ -322,14 +339,35 @@ void Realm<T>::constructor(ContextType ctx, ObjectType this_object, size_t argc,
     ensure_directory_exists_for_file(config.path);
 
     SharedRealm realm = realm::Realm::get_shared_realm(config);
-    auto delegate = new RealmDelegate<T>(realm, Context<T>::get_global_context(ctx));
-
-    if (!realm->m_binding_context) {
-        realm->m_binding_context.reset(delegate);
+    
+    // Fix for datetime -> timestamp conversion
+    if (realm->config().upgrade_initial_version != realm->config().upgrade_final_version &&
+        realm->config().upgrade_initial_version < 5) {
+        // any versions earlier than file format 5 are stored as milliseconds and need to be converted to the new format
+        for (auto object_schema : *realm->config().schema) {
+            auto table = ObjectStore::table_for_object_type(realm->read_group(), object_schema.name);
+            for (auto property : object_schema.properties) {
+                if (property.type == PropertyTypeDate) {
+                    if (!realm->is_in_transaction()) {
+                        realm->begin_transaction();
+                    }
+                    
+                    for (size_t row_index = 0; row_index < table->size(); row_index++) {
+                        auto milliseconds = table->get_timestamp(property.table_column, row_index).get_seconds();
+                        table->set_timestamp(property.table_column, row_index, Timestamp(milliseconds / 1000, (milliseconds % 1000) * 1000000));
```

Comment:
I think we should do some kind of detection of the order of magnitude. I could see cases where somebody converts a Cocoa/Android app to RN, or just adds some RN code to an existing Cocoa/Android app, and this messes up the dates.